source/ControlBackend.h

```diff
@@ -64,6 +62,17 @@ class ControlBackend final : public ControlBackendInterface
                 GST_ERROR("Unable to register client");
                 return;
             }
+
+            m_logToGstHandler = std::make_shared<firebolt::rialto::LogToGstHandler>();
```

Comment:
This still means we register after ipc has connected. I think what Adam was saying is instead (or aswell as) a registerLogHandler api, we have the ability to pass the log handler to:
m_control = IControlFactory::createFactory()->createControl();

We would also have to make changes to the way our internal ClientController connects to IPC, moving it out of the constructor as this object is created before we even create the Control object.

See what adam thinks.

Reply:
Yes, this is what I meant.

Reply:
I have moved the log handler registration to rialto_mse_base_sink_class_init() within source/RialtoGStreamerMSEBaseSink.cpp. Also I have delayed the creation of the singleton IpcClient class until the factory is called. This seems sufficient to start logging before creation of IpcClient...

I tested this by temporarily adding this line...

std::shared_ptr<IChannelFactory> IChannelFactory::createFactory()
{
+    RIALTO_IPC_LOG_INFO("@@@@ Create IPC factory");
     std::shared_ptr<IChannelFactory> factory;

And it appears in the output of my UK sky box's log...

root@skyxione:/tmp# journalctl -f | egrep 'rialto .* M:.* F:'
Jan 30 15:31:27 skyxione com.sky.as.apps_WebAppCobalt[2318]: <stderr> 0:00:17.308382278    18 0xa6fb6f38 DEBUG                 rialto LogToGstHandler.cpp:73:log: M:MediaPipelineCapabilities.cpp F:MediaPipelineCapabilities L:67 > entry:
Jan 30 15:31:27 skyxione com.sky.as.apps_WebAppCobalt[2318]: <stderr> 0:00:17.308593999    18 0xa6fb6f38 INFO                  rialto LogToGstHandler.cpp:69:log: M:IpcChannelImpl.cpp F:createFactory L:74 > @@@@ Create IPC factory